### PR TITLE
Add missing tests for datepicker/calendar

### PR DESCRIPTION
**Pull request by mhchen**

### CODE REVIEW

**Comment by mhchen** [comment]
@jelbourn Please review at your convenience

**Comment by ThomasBurleson** [comment]
@mhchen - see two (2) failed tests locally:

```console
[11:27:21] Running unit tests on unminified source.
Firefox 31.0.0 (Mac OS X 10.9.0) md-date-picker should update the internal input placeholder FAILED
	Expected 'undefined' to be ''.
	@/Users/thomasburleson/Documents/projects/Google/projects/material/src/components/datepicker/datePicker.spec.js:91:5

PhantomJS 1.9.8 (Mac OS X 0.0.0) md-date-picker should update the internal input placeholder FAILED
	Expected 'undefined' to be ''.
	Error: Expected 'undefined' to be ''.
	    at /Users/thomasburleson/Documents/projects/Google/projects/material/src/components/datepicker/datePicker.spec.js:91
PhantomJS 1.9.8 (Mac OS X 0.0.0): Executed 608 of 613 (1 FAILED) (skipped 5) (21.336 secs / 21.281 secs)
Firefox 31.0.0 (Mac OS X 10.9.0): Executed 608 of 613 (1 FAILED) (skipped 5) (7.542 secs / 7.395 secs)
TOTAL: 2 FAILED, 1214 SUCCESS
```

> Also can you squash the commits please.

**Comment by jelbourn** [comment]
I see the same failures when I run `gulp karma` locally.

File: src/components/datepicker/calendar.spec.js [hunk 1]
```diff
@@ -94,6 +94,29 @@ describe('md-calendar', function() {
...
   beforeEach(inject(function($injector) {
+    jasmine.addMatchers({
+      toBeSameDayAs: function() {
+        return {
+          compare: function(actual, expected) {
+            var results = {
+              pass: dateUtil.isSameDay(actual, expected)
+            };
+
+            var negation = !results.pass ? "" : "not ";
```
Comment (jelbourn) [inline src/components/datepicker/calendar.spec.js]: Prefer single-quotes for consistency.

File: src/components/datepicker/datePicker.js [hunk 2]
```diff
@@ -214,14 +214,15 @@
       self.closeCalendarPane();
     });
 
-    self.inputElement.addEventListener('input', angular.bind(self, self.resizeInputElement));
+    var angularElement = angular.element(self.inputElement);
```
Comment (jelbourn) [inline src/components/datepicker/datePicker.js]: ngElement works too

File: src/components/datepicker/datePicker.spec.js [hunk 3]
```diff
@@ -61,18 +61,93 @@ describe('md-date-picker', function() {
...
+      angular.element(controller.inputElement).triggerHandler('input');
+      $timeout.flush();
+      expect(controller.ngModelCtrl.$modelValue).toEqual(initialDate);
+    });
+
+    it('should add and remove the invalid class', function() {
+      controller.inputElement.value = '6/1/2015';
+      angular.element(controller.inputElement).triggerHandler('input');
+      $timeout.flush();
+      expect(controller.inputContainer.classList.contains('md-datepicker-invalid')).toEqual(false);
```
Comment (jelbourn) [inline src/components/datepicker/datePicker.spec.js]: You can use `toBe` for booleans.

**Comment by jelbourn** [comment]
Looks good other than those failures

**Comment by mhchen** [comment]
@ThomasBurleson @jelbourn Rebased, squashed, and fixed failing test.

I was unable to repro the failing test case, but I made the test more resilient to the empty case for the input element placeholder.

**Comment by jelbourn** [comment]
LGTM

**Pull request closed without merging**